napari/_qt/widgets/_tests/test_qt_play.py

```diff
@@ -160,18 +152,12 @@ def increment():
 
     view.dims.dims.events.current_step.connect(increment)
 
-    view.dims.play(0, 20)
     # wait for the thread to start before timing...
-    qtbot.waitSignal(view.dims._animation_thread.started, timeout=10000)
-    qtbot.wait(370)
+    with qtbot.waitSignal(view.dims._animation_thread.started, timeout=10000):
+        view.dims.play(0, 20)
+    qtbot.waitUntil(lambda: view.dims._frame >= 2)
     with qtbot.waitSignal(view.dims._animation_thread.finished, timeout=7000):
         view.dims.stop()
-    A = view.dims._frame
-    assert A >= 3
-
-    # make sure the stop button actually worked
-    qtbot.wait(150)
-    assert A == view.dims._frame
```

Comment:
I think waiting for `finished` is probably enough here. I know there is a `QTimer`, but in general we can't really guarantee what this value will be unless we can capture the last value in the function passed to `waitUntil`.